### PR TITLE
kvserver: move LoadAndReconcileReplicas to kvstorage

### PR DESCRIPTION
**Pull request by tbg**
Mechanical using GoLand.

Next steps are starting to add a datadriven testing framework
so that functionality can be exercised when it moves or is added here,
i.e. https://github.com/cockroachdb/cockroach/issues/93247.

Epic: none
Release note: None

### CODE REVIEW

**Comment by tbg** [comment]
This is identical to https://github.com/cockroachdb/cockroach/pull/93944#event-8077830744, which was closed by craig when I merged the `kvstorage` branch. Still working on my `git-machete` skills...

File: pkg/kv/kvserver/store.go [hunk 1]
```diff
@@ -1952,7 +1952,6 @@ func loadFullReplicaIDsFromDisk(
 // loadAndReconcileReplicas loads the Replicas present on this
 // store. It reconciles inconsistent state and runs validation checks.
 //
-// TODO(sep-raft-log): also load *uninitialized* Replicas.
```
Comment (pavelkalinnikov) [inline pkg/kv/kvserver/store.go:1955]: We're new setting `RaftReplicaID` for all ranges including uninitialized, right? Does this count as "load uninitialized Replicas", or there is still more work to do (creating the in-memory uninit replicas)?

**Review by pav-kv — APPROVED** [review]

File: pkg/kv/kvserver/store.go (hunk 1 above)
Comment (tbg) [reply pkg/kv/kvserver/store.go:1955]: We're setting it now. There still isn't a migration path for uninitialized replicas that were created before that change landed and then never touched (we do leak these routinely) but the migration path will be simply throwing away raft state belonging to such replicas.

I mentioned it here just to be sure: https://github.com/cockroachdb/cockroach/issues/93310#issuecomment-1362872896

**Comment by tbg** [comment]
bors r=pavelkalinnikov
TFTR!

**Pull request merged**